### PR TITLE
fix(appset): use topics for Gitlab SCM Provider (#13169)

### LINKED ISSUE #13169: SCM Provider: Use Gitlab project topics
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

Gitlab deprecated  [tags](https://docs.gitlab.com/ee/api/projects.html#get-single-project) for projects.

>([Deprecated](https://gitlab.com/gitlab-org/gitlab/-/issues/328226) in GitLab 14.0) The list of tags for a project; put array of tags, that should be finally assigned to a project. Use topics instead.

For compatibility reasons they decided to alias both fields, see the MR: <https://gitlab.com/gitlab-org/gitlab/-/merge_requests/60405>

**To Reproduce**

1. add a topic to your project
2. call API `gitlab.com/api/v4/projects/<PROJECT_ID>`

```json
...
  "tag_list": [
    "mytopic"
  ],
  "topics": [
    "mytopic"
  ],
...
```

**Expected behavior**

the correct field is used.

**Screenshots**

**Version**

```shell
Paste the output from `argocd version` here.
```

**Logs**

```
Paste any relevant application logs here.
```

### PR DESCRIPTION
**Pull request by tillepille**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 
* [x] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.

Please see [Contribution FAQs](https://argo-cd.readthedocs.io/en/latest/developer-guide/faq/) if you have questions about your pull-request.

## Description

Use Gitlab project topics since tags were long deprecated. More detail in #13169 

Closes  #13169

### CODE REVIEW

**Review by ishitasequeira — APPROVED** [review]
LGTM!!

File: applicationset/services/scm_provider/gitlab.go [hunk 1]
```diff
@@ -90,7 +90,7 @@ func (g *GitlabProvider) ListRepos(ctx context.Context, cloneProtocol string) ([
 				Repository:   gitlabRepo.Path,
 				URL:          url,
 				Branch:       gitlabRepo.DefaultBranch,
-				Labels:       gitlabRepo.TagList,
+				Labels:       gitlabRepo.Topics,
```
Comment (crenshaw-dev) [inline applicationset/services/scm_provider/gitlab.go:93]: Should we fall back to `TagList`, in case someone is using an ancient version of GitLab?

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

File: applicationset/services/scm_provider/gitlab.go (hunk 1 above)
Comment (tillepille) [reply applicationset/services/scm_provider/gitlab.go]: Yes, we probably should do that. 
I added a simple retry if the topics would be empty, hope that is fine.

**Review by crenshaw-dev — APPROVED** [review]
Thanks, @tillepille!

**Comment by crenshaw-dev** [comment]
@tillepille would you mind resolving the merge conflicts?

**Pull request merged**